jazz_codeq/index.js

```diff
@@ -0,0 +1,415 @@
+const request = require('request');
+const moment = require('moment');
+const async = require('async');
+const messages = require("./config/messages");
+const contextObj = require("./components/context.js"); //Import the context data.
+
+module.exports.handler = (event, context, cb) => {
+
+	//Initializations
+	let serviceContext = contextObj(event);
+	let errorHandler = errorHandlerModule();
+	let config = configObj(serviceContext);
+	let secretHandler = secretHandlerModule();
+	logger.init(serviceContext, context);
+
+	const DATE_FORMAT = 'YYYY-MM-DDThh:mm:ss';
+	const APPEND_ZEROES = '-0000';
+
+	try {
+
+		let pathString;
+		logger.info(serviceContext.resourcePath);
+		
+		getAPIPath(serviceContext.resourcePath)
+		.then(result => {
+			pathString = data.pathString;
+		}).catch(err => {
+			cb(JSON.stringify(errorHandler.throwInputValidationError(err.errorMessage)));
+		});
+
+		logger.info(pathString);
+
+		if (serviceContext && serviceContext.method && serviceContext.method === 'GET' && pathString === "codeq") {
+			logger.info("code quality service called.");
+
+			let required_fields = config.required_params.map(v => v.toLowerCase());
+			let query = getQuery(serviceContext);
+
+			let fromDate;
+			let toDate;
+
+			// validate required fields
+			let missing_required_fields = required_fields.filter(key => !query[key]);
+
+			if (missing_required_fields.length) {
+				message = messages.MISSING_FIELDS + missing_required_fields.join(", ");
+				logger.error(message);
+				return cb(JSON.stringify(errorHandler.throwInputValidationError(message)));
+			}
+
+			let metrics = getMetrics(query);
+
+			//if from and to dates are provided validate, else assign default dates
+			if (query.from) {
+				if (!moment(query.from, moment.ISO_8601, true).isValid()) {
+					return cb(JSON.stringify(errorHandler.throwInputValidationError(messages.INVALID_FROM_DATE)));
+				} else {
+					fromDate = moment(query.from).format(DATE_FORMAT) + APPEND_ZEROES;
+				}
+			} else {
+				fromDate = moment().subtract(1, "days").format(DATE_FORMAT) + APPEND_ZEROES;
+			}
+
+			if (query.to) {
+				if (!moment(query.to, moment.ISO_8601, true).isValid()) {
+					return cb(JSON.stringify(errorHandler.throwInputValidationError(messages.INVALID_TO_DATE)));
+				} else {
+					toDate = moment(query.to).format(DATE_FORMAT) + APPEND_ZEROES;
+				}
+			} else {
+				toDate = moment().format(DATE_FORMAT) + APPEND_ZEROES;
+			}
+
+			if (moment(fromDate).isAfter(toDate)) {
+				return cb(JSON.stringify(errorHandler.throwInputValidationError(messages.TO_EARLIER_THAN_FROM)));
+			}
+
+			let limit = query.limit ? query.limit : 100;
+			let offset = query.offset ? query.offset : 0;
+			let pageSize = limit + offset;
+
+			logger.info("metrics: " + metrics);
+			logger.info("from date: " + fromDate);
+			logger.info("to date: " + toDate);
+
+			let baseAuthToken, authToken;
+
+			decrypt_sonar_token()

```

Comment:
We are not using secret handler in OSS. Please remove this part.
@suryajak  please advise